editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -2445,6 +2445,51 @@ void VisualShaderEditor::_add_node(int p_idx, int p_op_idx, String p_resource_pa
 		vsnode->set_script(add_options[p_idx].script);
 	}
 
+	bool is_texture2d = false;
+	bool is_texture3d = false;
+	bool is_texture2d_array = false;
+	bool is_cubemap = false;
+	bool is_curve = false;
+	bool is_curve_xyz = false;
+	bool is_uniform = false;
+
+	{
+		VisualShaderNodeTexture *texture2d = Object::cast_to<VisualShaderNodeTexture>(vsnode.ptr());
+		if (texture2d) {
+			is_texture2d = true;
+		}
```

Comment:
Just a suggestion to make it shorter, but not sure if it's more readable:
```suggestion
	bool is_texture2d = (Object::cast_to<VisualShaderNodeTexture>(vsnode.ptr()) != nullptr);
```
etc.

(The `!= nullptr` is not necessary strictly speaking but here I guess it's good to make the intention explicit.)